packages/react/src/date-picker/DatePicker.tsx

```diff
@@ -11,18 +12,42 @@ import { usePopper } from 'react-popper';
 import { SelectSingleEventHandler } from 'react-day-picker';
 import { useClickOutside, useTernaryState } from '../core';
 import { CalendarSingle } from './Calendar';
-import { DateInput, DateInputProps } from './DatePickerInput';
+import { DateInput } from './DatePickerInput';
 import {
 	parseDate,
 	formatDate,
 	constrainDate,
 	transformValuePropToInputValue,
 } from './utils';
 
-type DatePickerInputProps = Omit<
-	DateInputProps,
-	'value' | 'onChange' | 'buttonRef' | 'buttonOnClick' | 'invalid'
```

Comment:
Rather than just adding a new value to this union type, I've decided to remove the dependency to `TextInput` and copy the types here like every other form component.

Yes this results in a bit more copy/paste, but it is makes the prop types so much easier to read.